engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -824,7 +824,7 @@ private DiskProfile allocateTemplatedVolume(Type type, String name, DiskOffering
         int volumesNumber = 1;
         List<DatadiskTO> templateAsIsDisks = null;
         String configurationId = null;
-        if (template.isDeployAsIs()) {
+        if (template.isDeployAsIs() && vm.getType() != VirtualMachine.Type.SecondaryStorageVm) {
```

Comment:
please note there is already a specific PR #4437 with this change

Reply:
This was added to verify the fix @sureshanaparti